extensions/interactions/base_test.py

```diff
@@ -212,81 +212,119 @@ def test_default_interactions_are_valid(self):
+            interaction_dir_contents = (
+                self._listdir_omit_ignored(interaction_dir))
 
-            optional_dirs_and_files_count = 0
+            interaction_dir_optional_dirs_and_files_count = 0
 
             try:
                 self.assertTrue(os.path.isfile(os.path.join(
-                    interaction_dir, '%sSpec.js' % interaction_id)))
-                optional_dirs_and_files_count += 1
+                    interaction_dir, 'protractor.js')))
+                interaction_dir_optional_dirs_and_files_count += 1
             except Exception:
                 pass
 
             try:
                 self.assertTrue(os.path.isfile(os.path.join(
                     interaction_dir,
-                    '%sValidationServiceSpec.js' % interaction_id)))
-                optional_dirs_and_files_count += 1
+                    '%sPredictionService.js' % interaction_id)))
+                interaction_dir_optional_dirs_and_files_count += 1
             except Exception:
                 pass
 
             try:
                 self.assertTrue(os.path.isfile(os.path.join(
-                    interaction_dir, '%sRulesServiceSpec.js' % interaction_id)))
-                optional_dirs_and_files_count += 1
+                    interaction_dir,
+                    '%sPredictionServiceSpec.js' % interaction_id)))
+                interaction_dir_optional_dirs_and_files_count += 1
             except Exception:
                 pass
 
+            self.assertEqual(
+                interaction_dir_optional_dirs_and_files_count + 4,
+                len(interaction_dir_contents)
+            )
+
+            py_file = os.path.join(interaction_dir, '%s.py' % interaction_id)
+            html_file = os.path.join(
+                interaction_dir, '%s.html' % interaction_id)
+
+            self.assertTrue(os.path.isfile(py_file))
+            self.assertTrue(os.path.isfile(html_file))
+
+            # Check that the directives subdirectory exists.
+            directives_dir = os.path.join(
+                interaction_dir, 'directives')
+            self.assertTrue(os.path.isdir(directives_dir))
+
+            # The directives directory should contain the following files:
+            #  Required:
+            #    * A JS file called {InteractionName}.js.
+            #    * A JS file called {InteractionName}ValidationService.js
+            #    * A HTML file called
+            #      {InteractionName}_interaction_directive.html
+            #    * A HTML file called
+            #      {InteractionName}_response_directive.html
+            #    * A HTML file called
+            #      {InteractionName}_short_response_directive.html
+            #  Optional:
+            #    * A JS file called {InteractionName}ValidationServiceSpecs.js
+            #    * A JS file called {InteractionName}RulesServiceSpecs.js
+
+            directives_dir_optional_dirs_and_files_count = 0
```

Comment:
We kept track of this above in order to verify that there are no extra files/folders, by verifying the count. Here, where do we check the count? (And if we don't need to, then maybe we don't need this variable.)

Reply:
I removed the count checking because it's not needed in this folder, but forgot to remove this variable. Done.